### PR TITLE
Make defaultState required when creating reducers

### LINKED ISSUE #18: Make defaultState non-optional
Since we have to return a non-undefined state in redux, we should probably make the `defaultState` parameter for `handleActions` required. Thoughts?

### PR DESCRIPTION
**Pull request by yangmillstheory**
Close #18.

See that issue and also https://github.com/reactjs/redux/issues/514 for background and discussion. 
***
**TL;DR**

We can prevent downstream reducer sanity errors [like this one](https://github.com/reactjs/redux/blob/master/src/combineReducers.js#L63) by throwing earlier.

### CODE REVIEW

**Comment by timche** [comment]
We are explicitly requiring it, so let's note it as breaking change, regardless of how Redux is handling it.

File: src/assertDefaultState.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import isUndefined from 'lodash/isUndefined';
+
+export default function assertDefaultState(defaultState, actionTypes) {
+  if (isUndefined(defaultState)) {
+    throw new Error(
+      `Expected defaultState for reducer handling ${actionTypes.join(', ')} to be defined`
+    );
+  }
+}
```
Comment (timche) [inline src/assertDefaultState.js]: I'd suggest to move this function to a folder named `utils` and name that file `assertions.js` or sth like this, where we can write more functions like this in a single file.

Comment (timche) [inline src/assertDefaultState.js]: Generally that's a good idea to extract the assertions 👍

**Comment by timche** [comment]
Btw, I've changed the base branch to `v1.0.0`, so we can merge some other stuff into `master` that's not supposed to be in `v1.0.0`.

File: src/assertDefaultState.js (hunk 1 above)
Comment (timche) [inline src/assertDefaultState.js]: Also, we will rewrite this later with `invariant`, correct?

Comment (yangmillstheory) [inline src/assertDefaultState.js]: Yes, or we may as well do it now.

**Comment by yangmillstheory** [comment]
I've updated the PR; it's been a while. We're now using `invariant` for runtime checks since https://github.com/acdlite/redux-actions/pull/141.

@JaKXz @timche can you take another look?

File: src/handleActions.js [hunk 2]
```diff
@@ -3,8 +3,13 @@ import ownKeys from './ownKeys';
...
 
 export default function handleActions(handlers, defaultState) {
-  const reducers = ownKeys(handlers).map(type => handleAction(type, handlers[type]));
+  const reducers = ownKeys(handlers).map(type =>
+    handleAction(
+      type,
+      handlers[type],
+      defaultState
+    )
+  );
```
Comment (yangmillstheory) [inline src/handleActions.js]: Relevant: https://github.com/acdlite/redux-actions/issues/108

**Pull request merged**